deployments/cluster/educator.nix

```diff
@@ -20,6 +20,18 @@ in
     4040        # Educator HTTP API
   ];
 
+  services.postgresql = {
+    enable = true;
+    authentication = ''
+      host all postgres 127.0.0.1/32 trust
+      host all postgres ::1/128      trust
```

Comment:
This `trust` looks suspicious :thinking:

Reply:
Looks fine to me 🤷‍♂️. What is suspicious about it?

Reply:
This allows any user on the machine to access any database. It's better to use ident, peer or password authentication to map access from specific users to specific databases. That way, if a service running as another user gets compromised, it does not gain access to the database.

Reply:
@Lucus16 Well, I would be glad to restrict access to the disciplina database to disciplina user, but I’m not quite sure how to obtain the username over a TCP connection. I am surprised that `ident` is actually still a thing, I thought it was some sort of a joke from before I was born, but, in any case, I have no clue how to use it – do we need to run some sort of ident daemon on the server or what?
An alternative would be to somehow connect to postgres over the unix socket and use `peer`, but I couldn’t figure out how to locate this socket reliably. Another alternative would be to use password authentication and, um, create a random password for the user on the first start and save it somewhere and then read in pre-start of the educator service? In my opinion, this all sounds like a little bit too much for a host running a single service.
If you have an idea, how to replace this `trust` authentication with reasonable effort, then let’s do it. (But, let’s keep in mind that we have to merge this PR tomorrow (today).)

Reply:
@kirelagin if you use unix sockets with postres (which is the default if you do not specify location or user or password), it will authenticate based on peer username (the 'peer' authentication method). The default should work fine.